spyder/plugins/explorer/widgets.py

```diff
@@ -851,7 +877,181 @@ def create_func(fname):
 
     def go_to_parent_directory(self):
         pass
-        
+
+    def copy_path(self, fnames=None, method="absolute"):
+        """Copy absolute or relative path to given file(s)/folders(s)."""
+        cb = QApplication.clipboard()
+        home_directory = self.fsmodel.rootPath()
+        if fnames is None:
+            fnames = self.get_selected_filenames()
+        if not isinstance(fnames, (tuple, list)):
+            fnames = [fnames]
+        fnames = [_fn.replace(os.sep, "/") for _fn in fnames]
+        if len(fnames) > 1:
+            if method == "absolute":
+                clipboard_files = ''.join('"' + _fn + '",' + '\n' for _fn in
+                                          fnames)
+            elif method == "relative":
+                clipboard_files = ''.join('"' +
+                                          osp.relpath(_fn, home_directory).
+                                          replace(os.sep, "/") + '",' +
+                                          '\n' for _fn in fnames)
+            clipboard_files = clipboard_files[:-2]
+        else:
+            if method == "absolute":
+                clipboard_files = fnames[0]
+            elif method == "relative":
+                clipboard_files = (osp.relpath(fnames[0], home_directory).
+                                   replace(os.sep, "/"))
+        cb.clear(mode=cb.Clipboard)
+        cb.setText(clipboard_files, mode=cb.Clipboard)
+
+    @Slot()
+    def copy_absolute_path(self):
+        """Copy absolute paths of named files/directories to the clipboard."""
+        self.copy_path(method="absolute")
+
+    @Slot()
+    def copy_relative_path(self):
+        """Copy relative paths of named files/directories to the clipboard."""
+        self.copy_path(method="relative")
+
+    @Slot()
+    def copy_file_clipboard(self, fnames=None):
+        """Copy file(s)/folders(s) to clipboard."""
+        if fnames is None:
+            fnames = self.get_selected_filenames()
+        if not isinstance(fnames, (tuple, list)):
+            fnames = [fnames]
+        try:
+            file_content = QMimeData()
+            file_content.setUrls([QUrl.fromLocalFile(_fn) for _fn in fnames])
+            file_content.setText('')
+            cb = QApplication.clipboard()
+            cb.clear(mode=cb.Clipboard)
+            cb.setMimeData(file_content, mode=cb.Clipboard)
+        except Exception as e:
+            QMessageBox.critical(self,
+                                 _('File/Folder Type Error'),

```

Comment:
* Please change `Type` here for `copy`, which I think is clearer.
* Also uncapitalize `Error`.